src/daemon/daemon.cpp

```diff
@@ -2053,6 +2055,37 @@ catch (const std::exception& e)
     status_promise->set_value(grpc::Status(grpc::StatusCode::INTERNAL, e.what(), ""));
 }
 
+void mp::Daemon::authenticate(const AuthenticateRequest* request,
+                              grpc::ServerWriterInterface<AuthenticateReply>* server,
+                              std::promise<grpc::Status>* status_promise)
+try
+{
+    mpl::ClientLogger<AuthenticateReply> logger{mpl::level_from(request->verbosity_level()), *config->logger, server};
+
+    auto stored_hash = MP_SETTINGS.get(mp::passphrase_key);
+
+    if (stored_hash.isNull())
+    {
+        return status_promise->set_value(
```

Comment:
```suggestion
        status_promise->set_value(
```

Reply:
Nope, this is intended and is done like this elsewhere in daemon code.  It's a neat trick to set something **and** to return at a given point for `void` functions.  Here, we want to return from the function and set the promise value.  If you don't like the look of this, it can be changed to something like:
```c++
        status_promise->set_value(...);
        return;
```

Reply:
D'uh.